src/index.js

```diff
@@ -91,33 +104,37 @@ if (fs.existsSync(fullPathToComponentDir)) {
 // Start by creating the directory that our component lives in.
 mkDirPromise(componentDir)
   .then(() => (
-    readFilePromiseRelative(templatePath)
+    generateTemplate(componentName, program.type, program.styling)
   ))
   .then(template => {
     logItemCompletion('Directory created.');
     return template;
   })
   .then(template => (
-    // Replace our placeholders with real data (so far, just the component name)
-    template.replace(/COMPONENT_NAME/g, componentName)
-  ))
-  .then(template => (
     // Format it using prettier, to ensure style consistency, and write to file.
-    writeFilePromise(filePath, prettify(template))
+    writeFilePromise(componentFilePath, prettify(template))
   ))
-  .then(template => {
+  .then(() => {
+    // Check whether we should make a CSS file too
+    if (hasCssFile(program.styling)) {
+      // Format it using prettier, to ensure style consistency, and write to file.
+      return writeFilePromise(cssFilePath, prettifyCss(cssTemplate))
+        .then(() => {
+          logItemCompletion('CSS file built and saved to disk.');
+        });
+    }
+  })
+  .then(() => {
     logItemCompletion('Component built and saved to disk.');
-    return template;
   })
-  .then(template => (
+  .then(() => (
     // We also need the `index.js` file, which allows easy importing.
     writeFilePromise(indexPath, prettify(indexTemplate))
   ))
-  .then(template => {
+  .then(() => {
     logItemCompletion('Index file built and saved to disk.');
-    return template;
   })
-  .then(template => {
```

Comment:
Good catch! Didn't realize I was pointlessly passing the template down :)

Reply:
I was confused for a few moments why we're doing this :))